java/src/jmri/jmrit/throttle/ThrottleFrame.java

```diff
@@ -546,24 +550,31 @@ private void editRendering() {
         }
     }
 
-    private boolean isEditMode = true;
-    private boolean willSwitch = false;
-
-    public void switchMode() {
+    public void setEditMode(boolean mode) {
+        if (mode == isEditMode)
+            return;
         if (isVisible()) {
-            if (isEditMode) {
+            if (!mode) {
                 playRendering();
             } else {
                 editRendering();
             }
-            isEditMode = !isEditMode;
+            isEditMode = mode;
             willSwitch = false;
         } else {
-            willSwitch = !willSwitch;
+            willSwitch = true;
         }
         throttleWindow.updateGUI();
     }
 
+    public boolean getEditMode() {
+        return isEditMode;
+    }
+
+    public void switchMode() {
```

Comment:
Please deprecate this method (it is, as you noted, ambiguous) by adding the following text above it:

```
    /**
      * @deprecated since 4.19.5; use {@link #setEditMode(boolean)} instead
      */
    @Deprecated
```